docs/DeveloperGuide.md

```diff
@@ -1,12 +1,406 @@
+
+![ListStep3.png](UML%2FList%2FListStep3.png)
+
+**Step 4.**. If the `printTable` method is called, it takes in an `ArrayList<Item> items` aas a parameter and prints out a table showing the name, UPC, quantity and price of all items in the inventory.
+
+
+
+
+
+
+
+### 2.2. Add
+The add command is mainly handled by the `AddCommand` and `AddParser` classes, which extends the `Command` and `Parser` 
+class respectively.
+#### 2.2.1. AddParser Class
+The `AddParser` class is a parser class that extends the `Parser` abstract class. It handles the "add" command by parsing 
+the user's input into separate input parameters using regular expressions. It then creates a new `Item` object with the 
+parsed parameters and creates a new `AddCommand` object with the `Inventory` object and `Item` object as parameters.
+
+The `AddParser` class has a constructor that takes in a `String` object representing the raw input from the user and an 
+`Inventory` object representing the inventory. It passes these parameters to the constructor of the `Parser` abstract class.
+
+The class contains an overridden `run()` method that first checks if the raw input is null or not. If it is null, it 
+throws a `MissingParametersException`. Otherwise, it uses a regular expression matcher to match the raw input to a 
+specific pattern. If the input doesn't match the pattern, it prints an error message and returns. If the input matches 
+the pattern, it creates a new `Item` object using the parsed parameters and creates a new `AddCommand` object using the 
+`Inventory` object and `Item` object. It then calls the `run()` method of the `AddCommand` object to execute the add command.
+![AddParser.png](UML%2FAdd%2FAddParser.png)
+
+#### 2.2.2. AddCommand Class
+The `AddCommand` class is a command class that extends the `Command` abstract class. It represents the command to add an 
+item to the inventory. It contains a constructor that takes in an `Inventory` object and an `Item` object as parameters. 
+The constructor sets the `Inventory` object to be the inventory of the `Command` class, and the `Item` object to be the item 
+to be added to the inventory. The class also contains a private method called `addItem()` that adds the `Item` object to 
+the inventory. The `addItem()` method checks if the item already exists in the inventory using its unique UPC 
+(Universal Product Code) code. If the item already exists, it prints a message stating that the item is a duplicate. 
+Otherwise, it adds the item to the inventory, updates the item name hash, and adds the item to the UPC code history. 
+If the `SessionManager`'s `autoSave` flag is enabled, it writes the current inventory to a file using the `SessionManager`.
+
+The `run()` method of the `AddCommand` class calls the `addItem()` method to execute the add command.
+![AddCommand.png](UML%2FAdd%2FAddCommand.png)
```

Comment:
not sure if `add(itemName, item)`, `writeSession(inventory)`, `writeCSV(inventory)` is correct, but I think that method calls should be `methodName(parameterName : parameterType)`?